cachito/workers/tasks/gomod.py

```diff
@@ -94,18 +94,22 @@ def _fail_if_parent_replacement_not_included(packages_json_data: PackagesData) -
                         f"Could not find parent Go module for package: {package['name']}"
                     )
 
-                dep_normpath = os.path.normpath(
-                    os.path.join(pkg_module_name, dependency["version"])
-                )
-                dep_module_name = match_parent_module(dep_normpath, modules)
-                if dep_module_name is None:
+                dep_module_name = match_parent_module(dependency["name"], modules)
+                check_module_path = dep_module_name
+                if not dep_module_name:
+                    dep_normpath = os.path.normpath(
+                        os.path.join(pkg_module_name, dependency["version"])
+                    )
+                    check_module_path = dep_normpath
+
+                if not check_module_path and dependency["name"] not in modules:
```

Comment:
I think this can be simpler:
```suggestion
                dep_name = dependency["name"]
                dep_normpath = os.path.normpath(os.path.join(pkg_module_name, dependency["version"]))
                dep_module_name = match_parent_module(dep_name, modules) or match_parent_module(dep_normpath, modules)

                if not dep_module_name:
```